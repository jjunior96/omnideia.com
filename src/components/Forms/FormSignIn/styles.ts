import { lighten } from 'polished';
import styled, { css } from 'styled-components';

export const Container = styled.div``;

export const ForgotPassword = styled.a`
  ${({ theme }) => css`
    text-align: right;
    display: block;

    color: ${theme.colors.black};

    font-size: ${theme.font.sizes.xxsmall};
    text-decoration: none;

    transition: color ${theme.transition.fast};

    &:hover {
      color: ${lighten(0.3, theme.colors.black)};
    }
  `}
`;
