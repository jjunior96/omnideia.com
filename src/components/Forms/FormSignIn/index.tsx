import { FiUser as UserIcon, FiLock as PasswordIcon } from 'react-icons/fi';

import Button from 'components/Button';
import Form from 'components/Form';
import Input from 'components/Input';

import * as S from './styles';

const FormSignIn = () => {
  return (
    <Form>
      <S.Container>
        <Input
          name="email"
          type="text"
          placeholder="E-mail"
          icon={<UserIcon />}
        />
        <Input
          name="password"
          type="password"
          placeholder="Password"
          icon={<PasswordIcon />}
        />

        <S.ForgotPassword href="#">Forgot your password?</S.ForgotPassword>

        <Button>Enter</Button>
      </S.Container>
    </Form>
  );
};

export default FormSignIn;
