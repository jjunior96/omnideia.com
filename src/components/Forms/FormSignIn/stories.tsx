import { Story, Meta } from '@storybook/react/types-6-0';

import FormSignIn from '.';

export default {
  title: 'Form/FormSignIn',
  component: FormSignIn
} as Meta;

export const Default: Story = (args) => (
  <div style={{ width: '36rem', margin: 'auto' }}>
    <FormSignIn {...args} />
  </div>
);
