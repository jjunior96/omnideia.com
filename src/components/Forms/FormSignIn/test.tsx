import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import FormSignIn from '.';

describe('<FormSignIn />', () => {
  it('should render the form correctly', () => {
    const { container } = renderWithTheme(<FormSignIn />);

    // Input email
    expect(screen.getByPlaceholderText(/e-mail/i)).toBeInTheDocument();

    // Input password
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();

    // Button de submit
    expect(screen.getByRole('button', { name: /enter/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render forgot text and link', () => {
    renderWithTheme(<FormSignIn />);

    // Texto do link de forgot password
    expect(screen.getByText(/forgot your password/i)).toBeInTheDocument();

    // Link de forgot password
    expect(
      screen.getByRole('link', { name: /forgot your password/i })
    ).toBeInTheDocument();
  });
});
