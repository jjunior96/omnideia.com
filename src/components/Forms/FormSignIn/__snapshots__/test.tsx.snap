// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render the form correctly 1`] = `
.c5 {
  width: 100%;
  color: #FAFAFA;
  background-color: #5855E9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.8rem;
  border: 0;
  padding: 0 1.6rem;
  padding-left: 2.4rem;
  font-weight: 600;
  font-size: 1.6rem;
  -webkit-transition: background-color 0.3s,box-shadow 0.2s,outline 0.1s;
  transition: background-color 0.3s,box-shadow 0.2s,outline 0.1s;
  outline: none;
  height: 5.6rem;
  font-size: 1.6rem;
}

.c5:focus {
  outline: 3px solid #b1aff5;
}

.c5:hover {
  background-color: #423ee6;
  box-shadow: 0 8px 8px rgba(0,0,0,0.1);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 1.4rem;
  margin: 0;
  padding: 0;
  line-height: 1;
  color: #0D0D0D;
}

.c1 span {
  display: none;
}

.c2 {
  background: #FAFAFA;
  color: #91AAB4;
  border-radius: 0.8rem;
  padding: 1.6rem 2rem;
  width: 100%;
  height: 56px;
  font-size: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #3E606F;
  cursor: default;
}

.c2:focus-within {
  outline: 3px solid #b1aff5;
}

.c2 input {
  color: #0D0D0D;
  background: transparent;
  outline: none;
  border: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 input::-webkit-outer-spin-button,
.c2 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c2 input[type='number'] {
  -moz-appearance: textfield;
}

.c2 input:-webkit-autofill,
.c2 input:-webkit-autofill:hover,
.c2 input:-webkit-autofill:focus {
  -webkit-text-fill-color: #3E606F;
  -webkit-box-shadow: #282f3a26 inset;
  box-shadow: #282f3a26;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.c2 input::-webkit-input-placeholder {
  color: #3E606F;
}

.c2 input::-moz-placeholder {
  color: #3E606F;
}

.c2 input:-ms-input-placeholder {
  color: #3E606F;
}

.c2 input::placeholder {
  color: #3E606F;
}

.c2 svg {
  margin-right: 1.6rem;
  color: #3E606F;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  padding: 0 2.4rem;
}

.c0 .c4 {
  margin-top: 1rem;
}

.c0 label {
  margin-top: 1.6rem;
}

.c0 button[type='submit'] {
  margin: 2.4rem 0;
}

.c3 {
  text-align: right;
  display: block;
  color: #0D0D0D;
  font-size: 1.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
}

.c3:hover {
  color: #5a5a5a;
}

<form
  class="styles__Container-sc-1m7ukcm-0"
>
  <div
    class="c0"
  >
    <div
      class="styles__Container-sc-12pvpyf-0"
    >
      <div
        class="styles__Container-sc-1scclvv-0"
      >
        <label
          class="c1"
        >
           
          <span>
            *
          </span>
        </label>
        <div
          class="c2"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
          <input
            name="email"
            placeholder="E-mail"
            type="text"
          />
        </div>
      </div>
      <div
        class="styles__Container-sc-1scclvv-0"
      >
        <label
          class="c1"
        >
           
          <span>
            *
          </span>
        </label>
        <div
          class="c2"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="11"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="11"
            />
            <path
              d="M7 11V7a5 5 0 0 1 10 0v4"
            />
          </svg>
          <input
            name="password"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <a
        class="c3"
        href="#"
      >
        Forgot your password?
      </a>
      <button
        class="c4 c5"
        color="primary"
      >
        <div
          class="c6"
        >
          Enter
        </div>
      </button>
    </div>
  </div>
</form>
`;
